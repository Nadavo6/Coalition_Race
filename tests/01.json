{
    "parties": [
      {
        "name": "The Lannisters",
        "mandates": 50,
        "join_policy": "M"
      },
      {
        "name": "The Baratheons",
        "mandates": 30,
        "join_policy": "M"
      },
      {
        "name": "The Targaryens",
        "mandates": 20,
        "join_policy": "L"
      },
      {
        "name": "The Starks",
        "mandates": 20,
        "join_policy": "L"
      }
    ],
    "graph": [
      [0, 0, 3, 1],
      [0, 0, 1, 1],
      [3, 1, 0, 2],
      [1, 1, 2, 0]
    ],
    "agents": [
      {
        "party_id": 0,
        "selection_policy": "E"
      },
      {
        "party_id": 3,
        "selection_policy": "E"
      }
    ]
  }
  