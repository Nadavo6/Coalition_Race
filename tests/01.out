[
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 20,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 2,
                "weight": 3
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 1,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 2
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 1,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 50,
                "name": "The Lannisters",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "Waiting"
            },
            {
                "id": 2,
                "mandates": 20,
                "name": "The Targaryens",
                "state": "Waiting"
            },
            {
                "id": 3,
                "mandates": 20,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 20,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 2,
                "weight": 3
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 1,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 2
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 1,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 50,
                "name": "The Lannisters",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "Waiting"
            },
            {
                "id": 2,
                "mandates": 20,
                "name": "The Targaryens",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 20,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 20,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 2,
                "weight": 3
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 1,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 2
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 1,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 50,
                "name": "The Lannisters",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 20,
                "name": "The Targaryens",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 20,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 3
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 20,
                "parties": [
                    3
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 2,
                "weight": 3
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 1,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 2
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 1,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 50,
                "name": "The Lannisters",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 20,
                "name": "The Targaryens",
                "state": "CollectingOffers"
            },
            {
                "id": 3,
                "mandates": 20,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 3
            },
            {
                "id": 2,
                "partyId": 2
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 40,
                "parties": [
                    3,
                    2
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 2,
                "weight": 3
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 1,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 2
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 1,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 50,
                "name": "The Lannisters",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "CollectingOffers"
            },
            {
                "id": 2,
                "mandates": 20,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 20,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    },
    {
        "agents": [
            {
                "id": 0,
                "partyId": 0
            },
            {
                "id": 1,
                "partyId": 3
            },
            {
                "id": 2,
                "partyId": 2
            },
            {
                "id": 3,
                "partyId": 1
            }
        ],
        "coalitions": [
            {
                "mandates": 50,
                "parties": [
                    0
                ]
            },
            {
                "mandates": 70,
                "parties": [
                    3,
                    2,
                    1
                ]
            }
        ],
        "edges": [
            {
                "from": 0,
                "to": 2,
                "weight": 3
            },
            {
                "from": 0,
                "to": 3,
                "weight": 1
            },
            {
                "from": 1,
                "to": 2,
                "weight": 1
            },
            {
                "from": 1,
                "to": 3,
                "weight": 1
            },
            {
                "from": 2,
                "to": 0,
                "weight": 3
            },
            {
                "from": 2,
                "to": 1,
                "weight": 1
            },
            {
                "from": 2,
                "to": 3,
                "weight": 2
            },
            {
                "from": 3,
                "to": 0,
                "weight": 1
            },
            {
                "from": 3,
                "to": 1,
                "weight": 1
            },
            {
                "from": 3,
                "to": 2,
                "weight": 2
            }
        ],
        "parties": [
            {
                "id": 0,
                "mandates": 50,
                "name": "The Lannisters",
                "state": "Joined"
            },
            {
                "id": 1,
                "mandates": 30,
                "name": "The Baratheons",
                "state": "Joined"
            },
            {
                "id": 2,
                "mandates": 20,
                "name": "The Targaryens",
                "state": "Joined"
            },
            {
                "id": 3,
                "mandates": 20,
                "name": "The Starks",
                "state": "Joined"
            }
        ]
    }
]
